Sources/NIO/Bootstrap.swift

```diff
@@ -242,6 +242,21 @@ public final class ServerBootstrap {
     }
 }
 
+private extension Channel {
+    func registerAndDoSynchronously(_ body: @escaping (Channel) -> EventLoopFuture<Void>) -> EventLoopFuture<Void> {
+        // this is pretty delicate at the moment:
+        // `body` (which will `connect`) must be _synchronously_ follow `register`, otherwise in our current
```

Comment:
@weissi nit: the comment is not really correct anymore as it is not only for `connect`.